api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+                                                                Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove an annotation from the specified parameter.
+     *
+     * @param parameter      the parameter to remove the annotation from
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  callable which declares the parameter is not currently declared
+     *                                  on the type or if the parameter is not declared on either a
+     *                                  constructor or a method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromParameter(AnnotatedParameter<? super T> parameter,
+                                                     Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified parameter. If the callable which
+     * declares the parameter is not already present, it will be added. If the
+     * parameter is not already present on the callable, it will be added.
+     *
+     * @param parameter  the parameter to add the annotation to
+     * @param annotation the annotation to add
+     * @throws IllegalArgumentException if the annotation is null or if the
+     *                                  parameter is not declared on either a constructor or a method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToParameter(AnnotatedParameter<? super T> parameter, Annotation annotation);
+
+    /**
+     * Remove annotations from the type, and all of it's members. If an
+     * annotation of the specified type appears on the type declaration, or any
+     * of it's members it will be removed.
+     *
+     * @param annotationType the type of annotation to remove
+     * @throws IllegalArgumentException if the annotationType is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromAll(Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove the provided annotation from the type, and all of it's members. If the
+     * annotation appears on the type declaration, or any of it's members it will be removed.
+     *
+     * @param annotation the annotation to remove
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromAll(Annotation annotation);
+
+    /**
+     * Reads in from an existing AnnotatedType. Any elements not present are
+     * added. The javaClass will be read in. If the annotation already exists on
+     * that element in the configurator the read annotation will be used.
+     *
+     * @param type the type to read from
+     * @throws IllegalArgumentException if type is null
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(AnnotatedType<U> type);
```

Comment:
I don't think these read methods make sense. The configurator should be always initialized with some AT (i.e. the class structure must be set) and then users should only be allowed to modify annotations. Maybe I'm wrong but I don't understand the purpose of these methods. Also `BeforeBeanDiscovery.addAnnotatedType(String)` should have a param (Class/AT) the configurator could be initialized from. Otherwise users could try to use `.addAnnotatedType(String).removeFromAll()` -> BANG!